tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -1,32 +1,493 @@
+            options.Instance = TestConstants.AadInstance;
+            options.TenantId = TestConstants.TenantIdAsGuid;
+            options.ClientId = TestConstants.ClientId;
+        };
+
+        public WebAppExtensionsTests()
+        {
+            _configSection = GetConfigSection(_configSectionName);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void AddSignIn_WithConfigName(bool useServiceCollectionExtension)
+        {
+            var configMock = Substitute.For<IConfiguration>();
+            configMock.Configure().GetSection(_configSectionName).Returns(_configSection);
+              
+            var services = new ServiceCollection();
+            services.AddDataProtection();
+
+            if (useServiceCollectionExtension)
+                services.AddSignIn(configMock, _configSectionName, _oidcScheme, _cookieScheme, false);
+            else
+                new AuthenticationBuilder(services)
+                    .AddSignIn(configMock, _configSectionName, _oidcScheme, _cookieScheme, false);
+
+            var provider = services.BuildServiceProvider();
+
+            // Assert config bind actions added correctly
+            provider.GetRequiredService<IOptionsFactory<OpenIdConnectOptions>>().Create(_oidcScheme);
+            provider.GetRequiredService<IOptionsFactory<MicrosoftIdentityOptions>>().Create("");
+            configMock.Received(3).GetSection(_configSectionName);
+
+            AddSignIn_TestCommon(services, provider);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void AddSignIn_WithConfigActions(bool useServiceCollectionExtension)
+        {
+            var services = new ServiceCollection();
+            services.AddDataProtection();
+
+            if (useServiceCollectionExtension)
+                services.AddSignIn(_configureOidcOptions, _configureMsOptions, _oidcScheme, _cookieScheme, false);
+            else
+                new AuthenticationBuilder(services)
+                    .AddSignIn(_configureOidcOptions, _configureMsOptions, _oidcScheme, _cookieScheme, false);
+
+            var provider = services.BuildServiceProvider();
+
+            // Assert configure options actions added correctly
+            var configuredOidcOptions = provider.GetServices<IConfigureOptions<OpenIdConnectOptions>>().Cast<ConfigureNamedOptions<OpenIdConnectOptions>>();
+            var configuredMsOptions = provider.GetServices<IConfigureOptions<MicrosoftIdentityOptions>>().Cast<ConfigureNamedOptions<MicrosoftIdentityOptions>>();
+
+            Assert.Contains(configuredOidcOptions, o => o.Action == _configureOidcOptions);
+            Assert.Contains(configuredMsOptions, o => o.Action == _configureMsOptions);
+
+            AddSignIn_TestCommon(services, provider);
+        }
+
+        private void AddSignIn_TestCommon(IServiceCollection services, ServiceProvider provider)
```

Comment:
Please put all public methods before the private ones.

Reply:
I grouped these private methods right under the public ones that use them. I think it makes it more readable, instead of jumping to the end of the class each time. But I can add them to the end to be consistent with the rest of the project.

Reply:
i moved these around in the analyzer branch, so mentioned it for consistency. it came up as a warning, so i just implemented the suggestion instead of suppressing it.

---
In reply to: [417791959](https://github.com/AzureAD/microsoft-identity-web/pull/132#discussion_r417791959) [](ancestors = 417791959)